### PR TITLE
Devbranch

### PR DESCRIPTION
**Pull request by freerangeunicorn**
@schremsn 

- [ ] Code review for deployment

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (schremsn) [inline .vscode/settings.json:1]: recommend to add vscode settings and workspace to gitignore

File: Yogic.py [hunk 2]
```diff
@@ -1 +1,441 @@
...
+        return user
+
+def identity():
+    pass
+
+#init app
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+app.config['SQLALCHEMY_DATABASE_URI'] = "postgres://Mai:Codeordie2019@localhost/yogic" #environment file
```
Comment (schremsn) [inline Yogic.py:28]: for deployment this needs to be moved to env file

File: Yogic.py [hunk 3]
```diff
@@ -1 +1,441 @@
...
+
+db = SQLAlchemy(app)
+migrate = Migrate(app, db)
+manager = Manager(app)
+#setup the JWT Manager
+
+app.config['JWT_SECRET_KEY'] = 'bitcheswhocode199000302' #can I change the secret key again? #environment file
```
Comment (schremsn) [inline Yogic.py:35]: env file (see above)

File: Yogic.py [hunk 4]
```diff
@@ -1 +1,441 @@
...
+        #print(type(_teacher))
+    print(json.dumps(list_teacher))
+    return Response(json.dumps(list_teacher), mimetype='application/json')
+
+
+#sign up form in react will call this method
+@app.route('/api/teacher', methods=['POST'])
+
+def create_teacher():
+    for key, value in request.headers.items():
```
Comment (schremsn) [inline Yogic.py:135]: remove before deployment

File: Yogic.py [hunk 5]
```diff
@@ -1 +1,441 @@
...
+        
+    db.session.commit()
+    return jsonify({ }), 200
+
+    
+#signup form in React will call this method
+@app.route('/api/student', methods=['POST'])
+def create_student():
+    data = request.json
+    print(data)
```
Comment (schremsn) [inline Yogic.py:203]: remove, since it includes password

File: Yogic.py [hunk 6]
```diff
@@ -1 +1,441 @@
...
+        return response
+
+
+
+@app.route('/api/teacher', methods=['PUT'])
+@jwt_required
+def edit_teacher():
+    claim = get_jwt_identity()
+    data = request.json
+    teacher_id = claim.get('teacher')
```
Comment (schremsn) [inline Yogic.py:186]: will return null if it's a student, so you should return an unauthorized error
also applies to get/edit teacher/student

File: Yogic.py [hunk 7]
```diff
@@ -1 +1,441 @@
...
+        d_student.update({'id':_student.id})
+        d_student.update({'first_name': _student.first_name})
+        d_student.update({'last_name': _student.last_name})
+        list_student.append(d_student)
+    return json.dumps(list_student)
+
+@app.route('/api/bookclass', methods=['PUT'])
+@jwt_required
+def book():
+    api_key = 'e63418a078174be4d3ddd19b1b670397'
```
Comment (schremsn) [inline Yogic.py:235]: look into moving to env file

**Review by schremsn — APPROVED** [review]

File: Yogic.py (hunk 7 above)
Comment (freerangeunicorn) [reply Yogic.py:235]: env file on Heroku isn't a good approach. Instead, you can use its built-in support for environment variables, using heroku config:set <var> <value> or its web UI. Either way, you'll get a regular environment variable.

File: .vscode/settings.json (hunk 1 above)
Comment (freerangeunicorn) [reply .vscode/settings.json:1]: done 👍

File: Yogic.py (hunk 2 above)
Comment (freerangeunicorn) [reply Yogic.py:28]: done 👍

File: Yogic.py (hunk 7 above)
Comment (freerangeunicorn) [reply Yogic.py:235]: done

**Pull request closed without merging**